lib/index.js

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const assert = require('assert-plus');
+const bedrock = require('bedrock');
+const {util: {clone, BedrockError}} = bedrock;
+
+const packages = [];
+
+/**
+ * @module bedrock-package-manager
+ */
+
+/**
+ * Get package information.
+ *
+ * @param {Object} options - The options to use.
+ * @param {string} [options.alias] - The package alias.
+ * @param {string} [options.packageName] - The package name.
+ *
+ * @returns {PackageInfo} The package information.
+ * @throws {BedrockError} Will throw a `NotFoundError` if the package is not
+ *   found.
+ */
+exports.get = ({alias, packageName, type}) => {
+  assert.optionalString(alias, 'options.alias');
+  assert.optionalString(packageName, 'options.packageName');
+  assert.string(type, 'options.type');
+
+  let pkg;
+  if(alias) {
+    pkg = packages.find(p => (p.type === type && p.alias === alias));
+  }
+  if(packageName) {
+    pkg = packages.find(p => (p.type === type &&
+      p.packageName === packageName));
+  }
+
+  if(!pkg) {
+    throw new BedrockError('Package not found.', 'NotFoundError', {
+      alias,
+      httpStatusCode: 404,
+      packageName,
+      public: true,
+      type,
+    });
+  }
+  return clone(pkg);
```

Comment:
Why the clone here and elsewhere?

Reply:
Otherwise the caller has the ability to mutate the package info and potentially cause corruption/breakage.

Reply:
Same phenomenon we see with jsonld.documentLoader.

Reply:
I have doubt this is a real issue here.  But I guess if we find this is a performance issue we can change it later.

Reply:
As an alternative, we could in a `bedrock.start` listener do a deep freeze on all the package objects using https://www.npmjs.com/package/deep-freeze

Reply:
Actually, I think we can just freeze them during registration, doing it.